packages/rest/test/unit/coercion/utils.ts

```diff
@@ -0,0 +1,80 @@
+import {
+  OperationObject,
+  ParameterObject,
+  SchemaObject,
+} from '@loopback/openapi-v3-types';
+
+import {
+  ShotRequestOptions,
+  expect,
+  stubExpressContext,
+} from '@loopback/testlab';
+
+import {
+  PathParameterValues,
+  Request,
+  Route,
+  createResolvedRoute,
+  parseOperationArgs,
+  ResolvedRoute,
+} from '../../..';
+
+export function givenOperationWithParameters(params?: ParameterObject[]) {
+  return <OperationObject>{
+    'x-operation-name': 'testOp',
+    parameters: params,
+    responses: {},
+  };
+}
+
+export function givenRequest(options?: ShotRequestOptions): Request {
+  return stubExpressContext(options).request;
+}
+
+export interface TestArgs<T> {
+  expectedResult: T;
+  valueFromReq: string;
+  schema?: SchemaObject;
+  specConfig?: Partial<ParameterObject>;
+  caller: string;
+}
+
+export function givenResolvedRoute(
+  spec: OperationObject,
+  pathParams: PathParameterValues = {},
+): ResolvedRoute {
+  const route = new Route('get', '/', spec, () => {});
+  return createResolvedRoute(route, pathParams);
+}
+
+export async function testCoercion<T>(config: TestArgs<T>) {
+  try {
```

Comment:
I use a `try catch` here to append the caller's error stack(line number) to the assertion error